langchain4j-pinecone/src/main/java/dev/langchain4j/store/embedding/pinecone/PineconeEmbeddingStore.java

```diff
@@ -2,73 +2,79 @@
+import org.openapitools.client.model.IndexList;
+import org.openapitools.client.model.IndexModel;
 
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 
+import static dev.langchain4j.internal.Utils.isNullOrEmpty;
 import static dev.langchain4j.internal.Utils.randomUUID;
+import static dev.langchain4j.store.embedding.pinecone.PineconeHelper.metadataToStruct;
+import static dev.langchain4j.store.embedding.pinecone.PineconeHelper.structToMetadata;
+import static io.pinecone.commons.IndexInterface.buildUpsertVectorWithUnsignedIndices;
 import static java.util.Collections.emptyList;
 import static java.util.Collections.singletonList;
 import static java.util.Comparator.comparingDouble;
 import static java.util.stream.Collectors.toList;
 
 /**
  * Represents a <a href="https://www.pinecone.io/">Pinecone</a> index as an embedding store.
- * Current implementation assumes the index uses the cosine distance metric.
- * Does not support storing {@link dev.langchain4j.data.document.Metadata} yet.
+ * <p>Current implementation assumes the index uses the cosine distance metric.</p>
  */
 public class PineconeEmbeddingStore implements EmbeddingStore<TextSegment> {
 
     private static final String DEFAULT_NAMESPACE = "default"; // do not change, will break backward compatibility!
     private static final String DEFAULT_METADATA_TEXT_KEY = "text_segment"; // do not change, will break backward compatibility!
 
-    private final PineconeConnection connection;
+    private final Index index;
     private final String nameSpace;
     private final String metadataTextKey;
 
     /**
      * Creates an instance of PineconeEmbeddingStore.
      *
-     * @param apiKey      The Pinecone API key.
-     * @param environment The environment (e.g., "northamerica-northeast1-gcp").
-     * @param projectId   The ID of the project (e.g., "19a129b"). This is <b>not</b> a project name.
-     *                    The ID can be found in the Pinecone URL: https://app.pinecone.io/organizations/.../projects/...:{projectId}/indexes.
-     * @param index       The name of the index (e.g., "test").
-     * @param nameSpace   (Optional) Namespace. If not provided, "default" will be used.
-     * @param metadataTextKey   (Optional) The key to find the text in the metadata. If not provided, "text_segment" will be used.
+     * @param apiKey          The Pinecone API key.
+     * @param index           The name of the index (e.g., "test").
+     * @param nameSpace       (Optional) Namespace. If not provided, "default" will be used.
+     * @param metadataTextKey (Optional) The key to find the text in the metadata. If not provided, "text_segment" will be used.
+     * @param createIndex     (Optional) Whether to create index or not. If not provided "false" will be used
+     * @param indexParam      (Optional) Parameters to create Index, see {@link PineconeServerlessIndexParam} and {@link PineconePodIndexParam}
```

Comment:
```suggestion
     * @param indexConfig      (Optional) Configuration parameters to create an index, see {@link PineconeServerlessIndexConfig} and {@link PineconePodIndexConfig}
```
WDYT?

Reply:
These two parameters only make sense in this configuration
```java
.createIndex(true)
.indexConfig(PineconePodIndexConfig.builder()...build())
```

What about simplifying it to:
```java
.createIndex(PineconePodIndexConfig.builder()...build())
```
?

Reply:
Good point for me! Already simplifying it.